### PR TITLE
Improve logging when a remote host went down suddenly

### PR DESCRIPTION
**Pull request by VicDeo**
## Description
- Always init `$response` variable
- Check the exception code against curl connection timeout and  log a short warning instead of the exception in this case.

## Related Issue
- https://github.com/owncloud/enterprise/issues/3494

## Motivation and Context
Cleaner log file

## How Has This Been Tested?
If the host went down the following notice appears:
```
{
"reqId":"3EdDVjLQAdFPB3WKwPWW",
"level":3,
"user":"--",
"app":"PHP",
"method":"--",
"url":"--",
"message":"Undefined variable: response at owncloud_10.2.1\/lib\/private\/Files\/Storage\/DAV.php#282"
}
```
followed by the exception:
```
{
"reqId":"pBFrHaZTKmwAgQuPYn5C",
"level":3,
"time":"2019-08-23T14:15:07+02:00",
"remoteAddr":",
"user":"--",
"app":"no app in context",
"method":"--",
"url":"--",
"message":"Exception:
{
"Exception":"SabreHTTPClientException",
"Message":"Failed connect to hostname; Connection timed out",
"Code":7,
"Trace":"A_LONG_TRACE_HERE"
```

this PR removes the notice and replaces the exception entry with 
```
{
"reqId":"AqiNEOi2818nZryQoqRw",
"level":2,
"time":"2019-09-11T15:29:43+00:00",
"remoteAddr":"127.0.0.1",
"user":"admin",
"app":"no app in context",
"method":"GET","url":"\/oc\/remote.php\/webdav\/testor%20(2)","message":"Storage is not available due to the connection timeout to  host.down"}
```

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: lib/private/Files/Storage/DAV.php [hunk 1]
```diff
@@ -274,11 +275,18 @@ protected function propfind($path) {
...
 					$this->statCache->set($path, false);
-					return false;
+				} else {
+					$this->convertException($e, $path);
 				}
-				$this->convertException($e, $path);
 			} catch (\Exception $e) {
-				$this->convertException($e, $path);
+				if ($e->getCode() === \CURLE_COULDNT_CONNECT || $e->getCode() === \CURLE_OPERATION_TIMEDOUT) {
+					\OC::$server->getLogger()->warning(
```
Comment (PVince81) [inline lib/private/Files/Storage/DAV.php]: shouldn't we throw StorageNotAvailableException instead of ignoring the error ? or is this happening anyway ?

Comment (VicDeo) [reply lib/private/Files/Storage/DAV.php]: We just return `false` in this method and StorageNotAvailableException is thrown later:
```
{"reqId":"AqiNEOi2818nZryQoqRw","level":2,"time":"2019-09-11T15:29:43+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"no app in context","method":"GET","url":"\/remote.php\/webdav\/testor%20(2)","message":"Connection timeout to oc.ufz.de:443"}
{"reqId":"AqiNEOi2818nZryQoqRw","level":3,"time":"2019-09-11T15:29:43+00:00","remoteAddr":"127.0.0.1","user":"admin","app":"webdav","method":"GET","url":"\/remote.php\/webdav\/testor%20(2)","message":"Caused by: {\"Exception\":\"OCP\\\\Files\\\\StorageNotAvailableException\",\"Message\":\"\",\"Code\":1,\"Trace\":"..."
```

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by phil-davis** [comment]
To test manually:
- have a local and federated server
- share a folder from local to federated server
- accept the share on federated server
- stop the local server
- refresh the webUI on federated server

I get log messages like:
```
[Thu Oct  3 10:42:28 2019] remote root vanished
[Thu Oct  3 10:42:28 2019] Storage is not available due to the connection timeout to 172.17.0.1:8080
[Thu Oct  3 10:42:28 2019] Caused by: {"Exception":"OCP\\Files\\StorageNotAvailableException","Message":"","Code":1,"Trace":"#0 \/home\/phil\/federated\/core\/apps\/files_sharing\/lib\/External\/Storage.php(187): OCA\\Files_Sharing\\External\\Storage->checkStorageAvailability()\n#1 \/home\/phil\/federated\/core\/lib\/private\/Files\/Storage\/Wrapper\/Wrapper.php(378): OCA\\Files_Sharing\\External\\Storage->hasUpdated('', 1570077048)\n#2 \/home\/phil\/federated\/core\/lib\/private\/Files\/Storage\/Wrapper\/Availability.php(394): OC\\Files\\Storage\\Wrapper\\Wrapper->hasUpdated('', 1570077048)\n#3 \/home\/phil\/federated\/core\/lib\/private\/Files\/Storage\/Wrapper\/Wrapper.php(378): OC\\Files\\Storage\\Wrapper\\Availability->hasUpdated('', 1570077048)\n#4 \/home\/phil\/federated\/core\/lib\/private\/Files\/Storage\/Wrapper\/Wrapper.php(378): OC\\Files\\Storage\\Wrapper\\Wrapper->hasUpdated('', 1570077048)\n#5 \/home\/phil\/federated\/core\/lib\/private\/Files\/Cache\/Watcher.php(125): OC\\Files\\Storage\\Wrapper\\Wrapper->hasUpdated('', 1570077048)\n#6 \/home\/phil\/federated\/core\/lib\/private\/Files\/View.php(1392): OC\\Files\\Cache\\Watcher->needsUpdate('', Object(OC\\Files\\Cache\\CacheEntry))\n#7 \/home\/phil\/federated\/core\/lib\/private\/Files\/View.php(1430): OC\\Files\\View->getCacheEntry(Object(OCA\\Files_Trashbin\\Storage), '', 'qqq')\n#8 \/home\/phil\/federated\/core\/apps\/dav\/lib\/Connector\/Sabre\/ObjectTree.php(179): OC\\Files\\View->getFileInfo('\/fed1\/files\/qqq')\n#9 \/home\/phil\/federated\/core\/apps\/dav\/lib\/Tree.php(70): OCA\\DAV\\Connector\\Sabre\\ObjectTree->getNodeForPath('qqq')\n#10 \/home\/phil\/federated\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(966): OCA\\DAV\\Tree->getNodeForPath('files\/fed1\/qqq')\n#11 \/home\/phil\/federated\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(1649): Sabre\\DAV\\Server->getPropertiesIteratorForPath('files\/fed1\/qqq', Array, 1)\n#12 \/home\/phil\/federated\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/CorePlugin.php(363): Sabre\\DAV\\Server->generateMultiStatus(Object(Generator), false)\n#13 \/home\/phil\/federated\/core\/lib\/composer\/sabre\/event\/lib\/WildcardEmitterTrait.php(96): Sabre\\DAV\\CorePlugin->httpPropFind(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#14 \/home\/phil\/federated\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(464): Sabre\\DAV\\Server->emit('method:PROPFIND', Array)\n#15 \/home\/phil\/federated\/core\/lib\/composer\/sabre\/dav\/lib\/DAV\/Server.php(241): Sabre\\DAV\\Server->invokeMethod(Object(Sabre\\HTTP\\Request), Object(Sabre\\HTTP\\Response))\n#16 \/home\/phil\/federated\/core\/apps\/dav\/lib\/Server.php(326): Sabre\\DAV\\Server->start()\n#17 \/home\/phil\/federated\/core\/apps\/dav\/appinfo\/v2\/remote.php(31): OCA\\DAV\\Server->exec()\n#18 \/home\/phil\/federated\/core\/remote.php(165): require_once('\/home\/phil\/fede...')\n#19 {main}","File":"\/home\/phil\/federated\/core\/apps\/files_sharing\/lib\/External\/Storage.php","Line":227}
```
and the webUI says:
```
Storage is temporarily not available
```

Looks good.

**Pull request merged**